ui/app/components/send/gas-fee-display.js

```diff
@@ -0,0 +1,53 @@
+const Component = require('react').Component
+const h = require('react-hyperscript')
+const inherits = require('util').inherits
+const EthBalance = require('../eth-balance')
+const FiatValue = require('../fiat-value')
+const { getTxFeeBn } = require('../../util')
+
+module.exports = GasFeeDisplay
+
+inherits(GasFeeDisplay, Component)
+function GasFeeDisplay () {
+  Component.call(this)
+}
+
+GasFeeDisplay.prototype.render = function () {
+  const {
+    currentCurrency,
+    conversionRate,
+    gas,
+    gasPrice,
+    blockGasLimit,
+  } = this.props
+
+  const renderableCurrencies = {
```

Comment:
I would suggest refactoring both into own method and then do a switch-case, and return `<noscript />` as the default case. There are two reasons:

 - In the current implementation, if currentCurrency is `undefined`, the app will break in React v15 because the `render` function can no longer return undefined.
- This will create both components first, and the render the one based on props -- meaning that one of the component created is "wasted". This is more for the sake of code hygiene. The actual perf difference would be less than 5 ms.